core/src/sigverify_stage.rs

```diff
@@ -57,49 +55,61 @@ impl SigVerifyStage {
         verified_sender: CrossbeamSender<Vec<Packets>>,
         verifier: T,
     ) -> Self {
-        let thread_hdls = Self::verifier_services(packet_receiver, verified_sender, verifier);
-        Self { thread_hdls }
+        let thread_hdl = Self::verifier_services(packet_receiver, verified_sender, verifier);
+        Self { thread_hdl }
     }
 
     fn verifier<T: SigVerifier>(
-        recvr: &Arc<Mutex<PacketReceiver>>,
+        recvr: &PacketReceiver,
         sendr: &CrossbeamSender<Vec<Packets>>,
-        id: usize,
         verifier: &T,
     ) -> Result<()> {
-        let (batch, len, recv_time) = streamer::recv_batch(
-            &recvr.lock().expect("'recvr' lock in fn verifier"),
-            if perf_libs::api().is_some() {
-                RECV_BATCH_MAX_GPU
-            } else {
-                RECV_BATCH_MAX_CPU
-            },
-        )?;
+        let (mut batches, len, recv_time) = streamer::recv_batch(&recvr)?;
 
         let mut verify_batch_time = Measure::start("sigverify_batch_time");
-        let batch_len = batch.len();
-        debug!(
-            "@{:?} verifier: verifying: {} id: {}",
-            timing::timestamp(),
-            len,
-            id
-        );
-        sendr.send(verifier.verify_batch(batch))?;
+        let batches_len = batches.len();
+        debug!("@{:?} verifier: verifying: {}", timing::timestamp(), len,);
+        let mut received_ips = HashMap::new();
```

Comment:
nit, `received_ips` can be declared within `if`

Reply:
moved, thanks